### PR TITLE
Bl 1774 upgrade to rails 7

### PR DESCRIPTION
**Pull request by cdoyle-temple**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -2,7 +2,7 @@ CATALOG_COLLECTION="tul_cob-catalog-5.6.2.v0.13.4-2024-05-08_11-15-30"
 AZ_COLLECTION="tul_cob-az-7-prod"
 WEB_CONTENT_COLLECTION="tul_cob-web-4-prod"
 
-SOLR_URL="https://$SOLRCLOUD_USER:$SOLRCLOUD_PASSWORD@$SOLRCLOUD_HOST/solr/$CATALOG_COLLECTION"
```
Comment (dkinzer) [inline .env:5]: This needs not be hard coded

File: Gemfile [hunk 2]
```diff
@@ -2,19 +2,18 @@
...
-gem "rails", "6.1.7.7"
+gem "rails", "7.1.3.2"
 
 gem "actionpack-action_caching", "~> 1.2"
 gem "actionpack-page_caching", "~> 1.2"
 gem "alma", git: "https://github.com/tulibraries/alma_rb.git", branch: "main"
 gem "autoprefixer-rails"
 gem "awesome_print"
 gem "base64", "0.1.1"
-gem "bento_search"
```
Comment (dkinzer) [inline Gemfile:14]: Are we using a specific branch of this or main?

File: bin/dev [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+if ! command -v foreman &> /dev/null
+then
+  echo "Installing foreman..."
+  gem install foreman
+fi
+
+foreman start -f Procfile.dev "$@"
```
Comment (dkinzer) [inline bin/dev:9]: Is this part of rails 7?

File: spec/routing/routing_spec.rb [hunk 4]
```diff
@@ -4,7 +4,7 @@
 
 RSpec.describe "random 404", type: :request do
 
-  it "should redirect /errors/:not_found" do
+  xit "should redirect /errors/:not_found" do
```
Comment (dkinzer) [inline spec/routing/routing_spec.rb:7]: Why are we skipping this test?

File: .env (hunk 1 above)
Comment (cdoyle-temple) [reply .env:5]: mistake

File: bin/dev (hunk 3 above)
Comment (cdoyle-temple) [reply bin/dev:9]: yes

File: Gemfile (hunk 2 above)
Comment (cdoyle-temple) [reply Gemfile:14]: We are, checking why this isn't updated

File: spec/routing/routing_spec.rb (hunk 4 above)
Comment (cdoyle-temple) [reply spec/routing/routing_spec.rb:7]: It should not be. Making sure it passes.

File: Gemfile [hunk 5]
```diff
@@ -2,19 +2,18 @@
...
-gem "blacklight", "~> 7.37.0"
+gem "bento_search", git: "https://github.com/tulibraries/bento_search.git"
```
Comment (dkinzer) [inline Gemfile:13]: @cdoyle-temple , I mean shouldn't there be a branch reference here?  branch: "changes" or did you make changes directly to main/master branch?

Comment (cdoyle-temple) [reply Gemfile:13]: Correct, main branch

**Review by dkinzer — APPROVED** [review]

**Pull request merged**